heudiconv/dicoms.py

```diff
@@ -0,0 +1,507 @@
+            meta_info = bids_info
+        else:
+            # make nice with python 3 - same behavior?
+            meta_info = meta_info.copy()
+            meta_info.update(bids_info)
+            # meta_info = dict(meta_info.items() + bids_info.items())
+        try:
+            meta_info['TaskName'] = (re.search('(?<=_task-)\w+',
+                                               op.basename(infofile))
+                                     .group(0).split('_')[0])
+        except AttributeError:
+            pass
+    # write to outfile
+    with open(infofile, 'wt') as fp:
+        json.dump(meta_info, fp, indent=3, sort_keys=True)
+
+    return niftifile, infofile
+
+
+def embed_metadata_from_dicoms(bids, item_dicoms, outname, outname_bids,
+                               prov_file, scaninfo, tempdirs, with_prov,
+                               min_meta):
+    """
+    Enhance sidecar information file with more information from DICOMs
+
+    Parameters
+    ----------
+    bids
+    item_dicoms
+    outname
+    outname_bids
+    prov_file
+    scaninfo
+    tempdirs
+    with_prov
+    min_meta
+
+    Returns
+    -------
+
+    """
+    from nipype import Node, Function
+    tmpdir = tempdirs(prefix='embedmeta')
+
+    embedfunc = Node(Function(input_names=['dcmfiles', 'niftifile', 'infofile',
+                                           'bids_info', 'force', 'min_meta'],
+                              output_names=['outfile', 'meta'],
+                              function=embed_nifti),
+                     name='embedder')
+    embedfunc.inputs.dcmfiles = item_dicoms
+    embedfunc.inputs.niftifile = op.abspath(outname)
+    embedfunc.inputs.infofile = op.abspath(scaninfo)
+    embedfunc.inputs.min_meta = min_meta
+    if bids:
+        embedfunc.inputs.bids_info = load_json(op.abspath(outname_bids))
+    else:
+        embedfunc.inputs.bids_info = None
+    embedfunc.inputs.force = True
+    embedfunc.base_dir = tmpdir
+    cwd = os.getcwd()
+    try:
+        # MG - still a bug?
+        """
+        Ran into
+INFO: Executing node embedder in dir: /tmp/heudiconvdcm2W3UQ7/embedder
+ERROR: Embedding failed: [Errno 13] Permission denied: '/inbox/BIDS/tmp/test2-jessie/Wheatley/Beau/1007_personality/sub-sid000138/fmap/sub-sid000138_3mm_run-01_phasediff.json'
+while
+HEUDICONV_LOGLEVEL=WARNING time bin/heudiconv -f heuristics/dbic_bids.py -c dcm2niix -o /inbox/BIDS/tmp/test2-jessie --bids --datalad /inbox/DICOM/2017/01/28/A000203
+
+so it seems that there is a filename collision so it tries to save into the same file name
+and there was a screw up for that A
+
+/mnt/btrfs/dbic/inbox/DICOM/2017/01/28/A000203
+        StudySessionInfo(locator='Wheatley/Beau/1007_personality', session=None, subject='sid000138') 16 sequences
+        StudySessionInfo(locator='Wheatley/Beau/1007_personality', session=None, subject='a000203') 2 sequences
+
+
+in that one though
+        """
+        # if global_options['overwrite'] and op.lexists(scaninfo):
```

Comment:
also overwrite mode is disabled...?

Reply:
...and here

Reply:
It doesn't seem like did overwrite actually did anything here yet...